### DISCUSSION TITLE
Without open internet, how to send message

### DISCUSSION THREAD

**Discussion by shihab-geo** [bbernhard/signal-cli-rest-api#578] [Q&A] [answered]
Concerning the security issue, if there is no open internet available in the VM/POD where the "signal-cli-rest-api' docker container is running, how could I send messages?

If, the open internet is needed for the outbound call or sending messages then which {IP:PORT} or {Domain} firewall connectivity is needed from the VM/POD where the "signal-cli-rest-api' docker container is running for sending messages?

**Comment by bbernhard** [comment]
Sorry, I don't get what you mean. You obviously need to allow outbound connections, otherwise it isn't possible to send/receive messages to/from the Signal Server. The only thing you need to be careful about, is if you expose the REST API over the internet. In that case, I highly recommend to protect the API somehow (e.g with a reverse proxy).

**Reply by shihab-geo** [reply]
I meant, Without the internet, Could I send the signal messages?

**Reply by shihab-geo** [reply]
@bbernhard, I found that the Internet is required to send messages. 

Now could you please confirm whether the outbound call goes to this signal server **"https://signal.org/"** or Not?

Is my below message flow ok?

`SignalRestAPIDockerContainer -> https://signal.org[outbound Call] -> Sent Message Successful`

**Reply by bbernhard** [reply, marked as answer]
Yes, you need Internet and yes, the REST API directly talks to the Signal Servers.

**Reply by shihab-geo** [reply]
Thanks.
